File: zach-py/day4.py
# Elves need to clean up camp - each section has a unique ID
# Each elf is assigned to a range of section ID's
# Find and remove overlapping sections

file = open("input_day4.txt", "r")
c = 0
ass_pairs = 0
for line in file:

    comma_split = line.split(',')
        
    first_pair = comma_split[0]
    first_pair_split = first_pair.split('-')
    first_floor = first_pair_split[0]
    first_roof = first_pair_split[1]

    second_pair = comma_split[1]
    second_pair_split = second_pair.split('-')
    second_floor = second_pair_split[0]
    second_roof = second_pair_split[1]

    if (first_floor <= second_floor and first_roof >= second_roof):  # second is contained by first
        ass_pairs += 1
        
    elif (first_floor >= second_floor and first_roof <= second_roof):  #first is contained by second
        ass_pairs += 1

    print(ass_pairs)
file.close()
print(ass_pairs)

File: zach-py/Day4.java
import java.io.*;

public class Day4 {


    public static void main(String[] args) {

        int ass_pairs = 0;

        try {
            File file = new File("input4.txt");
            BufferedReader br = new BufferedReader(new FileReader(file));
            String line;
            while ((line = br.readLine()) != null) {
               
               String [] numbers = line.split(",");
               String [] firstPair = numbers[0].split("-");
               String [] secondPair = numbers[1].split("-");
               
               int first_floor = Integer.parseInt(firstPair[0]);
               int first_ceiling = Integer.parseInt(firstPair[1]);
               
               int second_floor = Integer.parseInt(secondPair[0]);
               int second_ceiling = Integer.parseInt(secondPair[1]);
                
               if (first_floor <= second_floor && first_ceiling >= second_ceiling) {
                    ass_pairs += 1;
               }
                else if (first_floor >= second_floor && first_ceiling <= second_ceiling) {
                    ass_pairs += 1;
                }
                
                //**** For Part 2 Add this parametr as well */
                else if ( second_floor <= first_ceiling && second_ceiling >= first_floor) {
                    ass_pairs += 1;
                }                    
               //System.out.println(first_floor +" " + first_ceiling + " " + second_floor + " " + second_ceiling);

            }
            br.close();
        } catch (IOException e) {
            e.printStackTrace();
        }

        System.out.println(" ass pairs: " + ass_pairs);


    }
   
}
